### ISSUE TITLE
App freezes on update

### ISSUE THREAD

**Issue by srameshr** [Microsoft/cordova-plugin-code-push#212]
This issue still exists on IOS and not on Android

Earlier reported here: https://github.com/Microsoft/cordova-plugin-code-push/issues/156

* cordova-plugin-code-push version: ```cordova-plugin-code-push 1.9.2-beta ``` 

* List of installed plugins:
```
code-push 1.9.0-beta "CodePushAcquisition"
com.sarriaroman.PhotoViewer 1.1.5 "PhotoViewer"
com.telerik.plugins.nativepagetransitions 0.6.5 "Native Page Transitions"
com.verso.cordova.clipboard 0.1.0 "Clipboard"
cordova-plugin-actionsheet 2.2.2 "ActionSheet"
cordova-plugin-camera 2.3.1 "Camera"
cordova-plugin-code-push 1.9.2-beta "CodePush"
cordova-plugin-compat 1.1.0 "Compat"
cordova-plugin-customurlscheme 4.2.0 "Custom URL scheme"
cordova-plugin-datepicker 0.9.0 "DatePicker"
cordova-plugin-device 1.0.1 "Device"
cordova-plugin-dialogs 1.1.1 "Notification"
cordova-plugin-facebook4 1.7.4 "Facebook Connect"
cordova-plugin-file 4.1.1 "File"
cordova-plugin-file-transfer 1.6.0 "File Transfer"
cordova-plugin-geolocation 2.4.1 "Geolocation"
cordova-plugin-request-location-accuracy 2.2.0 "Request Location Accuracy"
cordova-plugin-splashscreen 2.1.0 "Splashscreen"
cordova-plugin-whitelist 1.3.0 "Whitelist"
cordova-plugin-x-socialsharing 5.1.3 "SocialSharing"
cordova-plugin-x-toast 2.3.1 "Toast"
cordova-plugin-zip 3.1.0 "cordova-plugin-zip"
ionic-plugin-keyboard 2.2.1 "Keyboard"
org.apache.cordova.statusbar 0.1.10 "StatusBar"
phonegap-plugin-push 1.9.2 "PushPlugin"
```

* Cordova version: 6.5.0

* iOS/Android/Windows version:
```
  android 6.1.2
  ios 4.3.1
```

* Does this reproduce on a debug build or release build?: ```Staging and Release```

* Does this reproduce on a simulator, or only on a physical device?: Both

* Repro steps and/or stack trace:

(The more info the faster we will be able to address it!)

**Comment by max-mironov** [comment]
Hello @srameshr! That seems to be a known issue for old cordova-plugin-file version not using  background thread for some operations. 
This was fixed in cordova-plugin-file 4.3.0. So could you please update your version of cordova-plugin-file to the latest one and verify the results?

**Comment by srameshr** [comment]
@max-mironov Sorry! I posted this before going through the whole thread and forgot to close this after I found the issue with the plugin version.
